pkg/parser/parser.go

```diff
@@ -0,0 +1,121 @@
+package parser
+
+import (
+	"fmt"
+
+	"github.com/tjgurwara99/monkey-go/pkg/ast"
+	"github.com/tjgurwara99/monkey-go/pkg/token"
+)
+
+type parser struct {
+	tokens chan token.Token
+
+	current token.Token
+	peek    token.Token
+
+	errors []error
+}
+
+func New(t chan token.Token) *parser {
+	return &parser{
+		tokens: t,
+		errors: []error{},
+	}
+}
+
+func (p *parser) Errors() []error {
+	return p.errors
+}
+
+func (p *parser) nextToken() {
+	p.current = p.peek
+	p.peek = <-p.tokens
+}
+
+func (p *parser) ParseProgram() *ast.Program {
+	program := &ast.Program{
+		Statements: []ast.Statement{},
+	}
+
+	for !p.currentIs(token.EOF) {
+		stmt := p.ParseStatement()
+		if stmt != nil {
+			program.Statements = append(program.Statements, stmt)
+		}
+
+		p.nextToken()
+	}
+
+	return program
+}
+
+func (p *parser) ParseStatement() ast.Statement {
+	switch p.current.Type {
+	case token.LET:
+		return p.ParseLetStatement()
+	case token.RETURN:
+		return p.ParseReturnStatement()
+	default:
+		return nil
+	}
+}
+
+func (p *parser) ParseLetStatement() *ast.LetStatement {
+	stmt := &ast.LetStatement{Token: p.current}
+
+	if !p.expectPeek(token.IDENT) {
+		return nil
```

Comment:
I thought you'd be returning error or at least appending error into the parser itself since you're going the permissive parsing strategy.